scripts/ostrta.coffee

```diff
@@ -0,0 +1,85 @@
+# Description:
+#   One Script To Rule Them All
+#
+# Dependencies:
+#   None
+#
+# Configuration:
+#   None
+#
+# Commands:
+#   hubot listenFor <key> <percent%?> <response>
+#   hubot respondTo <key> <percent%?> <response>
+#
+# Notes:
+#
+# Author:
+#   maschall
+
+_ = require 'underscore'
+
+commandMap =
+  listenFor: "hear"
+  respondTo: "respond"
+
+brainKey = (command, key = "", topic = "") ->
+  "ostrta-#{command}#{topic}-#{key.replace(/\s/g, '_')}"
+
+add = (robot, key, value) ->
+  objects = fetch(robot, key)
+  if !objects
+    objects = []
+  objects.push value
+  remember(robot, key, objects)
+
+remember = (robot, key, value) ->
+  robot.brain.set(key, value)
+  
+fetch = (robot, key) ->
+  robot.brain.get(key)
+  
+percentFromString = (percentString) ->
+  parseInt(percentString, 10)
+  
+tellRobotWhatToDo = (robot, command, key) ->
+  robot[commandMap[command]](new RegExp("(#{key}(\\s|$))", 'i'), (msg) ->
+    response = fetch(robot, brainKey(command, key))
+    if response
+      percent = fetch(robot, brainKey(command, key, "Percent"))
+      if Math.random() <= (percent / 100.0)
+        msg.send response
+  )
+  
+loadAllCommands = ( robot, command ) ->
+  keys = fetch( robot, brainKey(command, key, "Keys") )
+  if keys
+    for key in keys
+      tellRobotWhatToDo( robot, command, key )
+  
+module.exports = (robot) ->
+  commands = _.keys(commandMap)
+  for command in commands
+    loadAllCommands( robot, command )
+
+  commandsRegex = commands.join('|')
+  robot.respond new RegExp("(#{commandsRegex}) (\"[^\"]+\"|[^\\s]+) ([\\d]+%\\s)?(.+)$"), (msg) ->
```

Comment:
named parts makes sense, how do you get the named parts out of the match object in coffeescript?